source/utils/paragraphHelper.py

```diff
@@ -0,0 +1,250 @@
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2022 NV Access Limited, Rob Meredith
+# This file may be used under the terms of the GNU General Public License, version 2 or later.
+# For more details see: https://www.gnu.org/licenses/gpl-2.0.html
+
+import api
+import speech
+import controlTypes
+import textInfos
+import tones
+import config
+from NVDAObjects.window.winword import WordDocumentTextInfo
+from NVDAObjects.window.winword import BrowseModeWordDocumentTextInfo
+from NVDAObjects.UIA import UIATextInfo
+from displayModel import EditableTextDisplayModelTextInfo
+
+MAX_LINES = 250  # give up after searching this many lines
+
+paragraphStyles = [
+	# values used for config.conf["paragraphNavigation"]["paragraphStyle"]
+	"auto",
+	"application",
```

Comment:
What is the difference between these? They appear to be the same in code.

If this is intended, I suggest using a [feature flag](https://github.com/nvaccess/nvda/blob/master/devDocs/featureFlags.md) and having default be application. I would also change "else" conditions to be explicit.

Reply:
@seanbudd  I'm confused. You suggested Auto as the default choice, so I added it. Yes, Auto and Application are the same. I don't think this is an option looking for a Feature Flag. Not sure where to go from here.

Reply:
I suggested auto before the concept of the generic "feature flag" was introduced.

The intention of adding "auto" is so that we can consider changing the default in the future.
This is the intended purpose of the feature flag concept.

What makes you think a feature flag is inappropriate in this case?

Reply:
@seanbudd So if I remove Auto, do we still need a feature flag? I'm still a bit weak on feature flags and how to use them. It seems like Application is the obvious choice for the default.

Reply:
The feature flag approach includes a "Default" option, which you should map to application.
This is to replace "Auto"

Reply:
@seanbudd I started implementation of the feature flag, but it appears there is an issue in NVDA when handling feature flagss which are not of type BoolFlag. Can you look at my branch and let me know if the error is on your end or mine? Error follows:
DEBUG - config.featureFlag._validateConfig_featureFlag (08:00:32.583) - MainThread (6896):
Validating feature flag: DEFAULT, optionsEnum: ParagraphNavigationFlag, behaviorOfDefault: APPLICATION
ERROR - unhandled exception (08:00:32.583) - MainThread (6896):
Traceback (most recent call last):
  File "gui\settingsDialogs.py", line 4167, in onCategoryChange
    super(NVDASettingsDialog,self).onCategoryChange(evt)
  File "gui\settingsDialogs.py", line 659, in onCategoryChange
    self._doCategoryChange(newIndex)
  File "gui\settingsDialogs.py", line 641, in _doCategoryChange
    newCat = self._getCategoryPanel(newCatId)
  File "gui\settingsDialogs.py", line 569, in _getCategoryPanel
    panel = cls(parent=self.container)
  File "gui\settingsDialogs.py", line 327, in __init__
    self._buildGui()
  File "gui\settingsDialogs.py", line 337, in _buildGui
    self.makeSettings(self.settingsSizer)
  File "gui\settingsDialogs.py", line 2563, in makeSettings
    conf=config.conf
  File "gui\guiHelper.py", line 332, in addLabeledControl
    labeledControl = LabeledControlHelper(parent, labelText, wxCtrlClass, **kwargs)
  File "gui\guiHelper.py", line 205, in __init__
    self._ctrl = WxCtrlWithEnableEvnt(parent, **kwargs)
  File "gui\nvdaControls.py", line 440, in __init__
    for value in self._optionsEnumClass
  File "gui\nvdaControls.py", line 441, in <dictcomp>
    if value != self._optionsEnumClass.DEFAULT
  File "utils\displayString.py", line 53, in displayString
    return self._displayStringLabels[self]
TypeError: 'method' object is not subscriptable

Reply:
_displayStringLabels is missing a property decorator: https://github.com/nvaccess/nvda/pull/13798/files#r957911881